manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/services/AbstractConnectorServiceTest.java

```diff
@@ -0,0 +1,167 @@
+import static com.redhat.service.smartevents.manager.v2.TestConstants.DEFAULT_CONNECTOR_TYPE_ID;
+import static com.redhat.service.smartevents.manager.v2.TestConstants.DEFAULT_CUSTOMER_ID;
+import static com.redhat.service.smartevents.manager.v2.TestConstants.DEFAULT_ORGANISATION_ID;
+import static com.redhat.service.smartevents.manager.v2.TestConstants.DEFAULT_USER_NAME;
+import static com.redhat.service.smartevents.manager.v2.utils.Fixtures.createReadyBridge;
+import static com.redhat.service.smartevents.manager.v2.utils.StatusUtilities.getManagedResourceStatus;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public abstract class AbstractConnectorServiceTest {
+
+    @Inject
+    BridgeDAO bridgeDAO;
+
+    @Inject
+    DatabaseManagerUtils databaseManagerUtils;
+
+    public abstract ConnectorService getConnectorService();
+
+    public abstract ConnectorDAO getConnectorDAO();
+
+    public abstract ConnectorType getConnectorType();
+
+    @BeforeEach
+    public void cleanUp() {
+        databaseManagerUtils.cleanUpAndInitWithDefaultShard();
+    }
+
+    @Test
+    void testCreateConnectorWhenBridgeNotActive() {
+        Bridge bridge = Fixtures.createAcceptedBridge(DEFAULT_BRIDGE_ID, DEFAULT_BRIDGE_NAME);
+        bridgeDAO.persist(bridge);
+
+        ConnectorRequest request = new ConnectorRequest();
+        request.setName(DEFAULT_CONNECTOR_NAME);
+        assertThatExceptionOfType(BridgeLifecycleException.class)
+                .isThrownBy(() -> getConnectorService().createConnector(bridge.getId(), bridge.getCustomerId(), bridge.getOwner(), bridge.getOrganisationId(), request));
+    }
+
+    @Test
+    void testCreateConnectorWhenBridgeDoesNotExist() {
+        ConnectorRequest request = new ConnectorRequest();
+        request.setName(DEFAULT_CONNECTOR_NAME);
+        assertThatExceptionOfType(ItemNotFoundException.class)
+                .isThrownBy(() -> getConnectorService().createConnector("id that does not exist", DEFAULT_CUSTOMER_ID, DEFAULT_USER_NAME, DEFAULT_ORGANISATION_ID, request));
+    }
+
+    @Test
+    void testCreateConnectorWhenConnectorWithSameNameAlreadyExists() {
+        Bridge bridge = Fixtures.createReadyBridge(DEFAULT_BRIDGE_ID, DEFAULT_BRIDGE_NAME);
+        bridgeDAO.persist(bridge);
+
+        Connector connector = Fixtures.createConnector(bridge, getConnectorType());
+        getConnectorDAO().persist(connector);
+
+        ConnectorRequest request = new ConnectorRequest();
+        request.setName(DEFAULT_CONNECTOR_NAME);
+        assertThatExceptionOfType(AlreadyExistingItemException.class)
+                .isThrownBy(() -> getConnectorService().createConnector(bridge.getId(), bridge.getCustomerId(), bridge.getOwner(), bridge.getOrganisationId(), request));
+    }
+
+    @Test
+    void testCreateConnectorWhenOrganizationHasNoQuota() {
```

Comment:
How does this test work?

I don't see anywhere you're mocking the quota limit or something to make `NoQuotaAvailable` be thrown.

The limit set for tests is 100.

Reply:
It is testing an organisation with no quota (by default it is `0` for organisations not included in the configuration of the quota)

Reply:
wops, I had a typo, now it is implemented the right way